packages/iframe-coordinator/src/HostRouter.ts

```diff
@@ -26,7 +31,7 @@ export class HostRouter {
   public getClientTarget(route: string): ClientTarget | null {
     let clientTarget: ClientTarget | null = null;
 
-    this._clients.forEach((client) => {
+    for (let client of this._clients) {
```

Comment:
Noticed this issue as well.
It was continuing to loop after finding a matching client, causing it to seem like it was searching back to front.
Now it breaks the loop after finding a matching client and returns the first match.